core/utils/bits.h

```diff
@@ -163,6 +163,26 @@ static inline void MaskBytes(uint8_t *buf, uint8_t const *mask,
   }
 }
 
+// TODO(melvinw): it would be nice if MaskTopN() supported blobs.
+// Returns a mask for the first `n` bits of an integral type `T`, i.e. starting
+// from the most significant bit toward the least significant. Returns a `T`
+// with all bits set if `n` is greater than the number of bits in `T`.
+// For example: MaskTopN<uint32_t>(9) will return 0xF8000000
```

Comment:
Occurs to me on rescan, 9 is `0xFF800000`. `0xF8000000` is what you get for 5...

Reply:
Doh! Good catch

Reply:
Fixed and rebase